driver/driver.go

```diff
@@ -184,20 +204,26 @@ func defaultOptions() *options {
 
 // Return a retry strategy with jittered exponential backoff, capped at the
 // given amount of time.
-func driverConnectionRetryStrategy(factor, cap time.Duration) strategy.Strategy {
+func driverConnectionRetryStrategy(factor, cap time.Duration, limit uint) []strategy.Strategy {
```

Comment:
I'd suggest to rename the function to:

```driverConnectionRetryStrategies```

(plural), since it's now returning a slice.